### PR TITLE
Updates port validation to restrict to valid port numbers 1-65535

### LINKED ISSUE #21556: Add database wizard is overzealous with port validation
Expected behavior: Database port should be able to be any number of digits (some hosted DB ports are 5 digits)

Current behavior: Port is restricted to 4 digits

image

Bug details:
- Teleport version: 12.0.1

### PR DESCRIPTION
**Pull request by stevenGravy**
Current version restricts to 4 digits.  Valid port ranges should be 1-65535.
Addresses #21556

### CODE REVIEW

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx [hunk 1]
```diff
@@ -337,14 +337,15 @@ const CreateDatabaseDialog = ({
   );
 };
 
-// PORT_REGEXP only allows digits with length 4.
-export const PORT_REGEX = /^\d{4}$/;
+// only allows digits with valid port range 1-65535. Not using RegEx as more effienct to use logic comparsions.
```
Comment (zmb3) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx:340]:
```suggestion
// only allows digits with valid port range 1-65535
```

This gets rid of a typo in effienct, and removes unnecessary info.

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx [hunk 2]
```diff
@@ -337,14 +337,15 @@ const CreateDatabaseDialog = ({
...
 const requirePort = value => () => {
-  const isValidId = value.match(PORT_REGEX);
+  const numberValue = Number(value);
```
Comment (hatched) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx:342]:
```suggestion
  const numberValue = parseInt(value);
```

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx (hunk 1 above)
Comment (rudream) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx:340]:
```suggestion
// Only allows digits with valid port range 1-65535. Not using RegEx as more efficient to use logic comparisons.
```

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx [hunk 3]
```diff
@@ -337,14 +337,15 @@ const CreateDatabaseDialog = ({
...
+  const isValidId =
```
Comment (rudream) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx:343]:
```suggestion
  const isValidPort =
```

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx [hunk 4]
```diff
@@ -337,14 +337,15 @@ const CreateDatabaseDialog = ({
...
+    Number.isInteger(numberValue) && numberValue >= 1 && numberValue <= 65535;
   if (!isValidId) {
```
Comment (hatched) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx:345]:
```suggestion
  const isValidPort = numberValue >= 1 && numberValue <= 65535;
  if (!isValidPort) {
```

**Review by rudream — APPROVED** [review]
With minor suggestions

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx (hunk 2 above)
Comment (zmb3) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx:342]: You probably want a try/catch here, as `Number()` can throw.

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx [hunk 5]
```diff
@@ -337,14 +337,15 @@ const CreateDatabaseDialog = ({
...
     return {
       valid: false,
-      message: 'port must be 4 digits',
+      message: 'port must be a valid port number',
```
Comment (hatched) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx:348]:
```suggestion
      message: 'Endpoint Port must be a valid port (1 - 65535).',
```

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx (hunk 4 above)
Comment (stevenGravy) [reply web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx]: addressed @hatched

**Review by hatched — APPROVED** [review]

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx (hunk 2 above)
Comment (stevenGravy) [reply web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx]: @hatched the parseInt should make the try/catch unnecessary right?

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx [hunk 6]
```diff
@@ -337,14 +337,15 @@ const CreateDatabaseDialog = ({
...
+// Only allows digits with valid port range 1-65535.
+const requirePort = (value: string) => () => {
+  const numberValue = parseInt(value);
+  const isValidPort =
+    Number.isInteger(numberValue) && numberValue >= 1 && numberValue <= 65535;
+  if (!isValidPort) {
     return {
       valid: false,
-      message: 'port must be 4 digits',
+      message: 'Endpoint Port must be a valid port (1 - 65535).',
```
Comment (kimlisa) [inline web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx:348]: the port input box width is quite small, so this err message will probably cause things to wrap and jump, what do you think about these messages:

`invalid port (1-65535)` or
`invalid range (1-65535)` 

if none of these is clear enough we might need to increase the port width

**Review by kimlisa — COMMENTED** [review]
thank you for the fix!

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx (hunk 6 above)
Comment (stevenGravy) [reply web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx]: done, it fits! 
image

**Review by kimlisa — APPROVED** [review]

File: web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx (hunk 6 above)
Comment (stevenGravy) [reply web/packages/teleport/src/Discover/Database/CreateDatabase/CreateDatabase.tsx]: please approve if you are good @zmb3

**Review by zmb3 — APPROVED** [review]

**Pull request merged**